Updating gaussian converter

anbenali opened the pull request:
Fix small bug in converter for  All electron orbital correction when … using HDF5. Behavior was duplicating CuspInfo line

markdewing commented:
okay to test

prckent commented:
After discussing with @anbenali , merge deferred for additional updates.

anbenali commented:
I am done with this part. Ready to merge. Rest of commits will be part of different branch.

prckent commented on src/QMCTools/convert4qmc.cpp:
```diff
@@ -55,10 +55,23 @@ int main(int argc, char **argv)
+  std::string prefix(argv[2]);
+
+
+  std::string delimiter =".out";
```
I think we could improve the handling here. Would it be simpler to add the requirement that the suffix actually is *.out ? Is this completely standard for GAMESS output?

anbenali replied:
this will pick anything that has .out* (so out or output will be taken)

prckent commented on src/QMCTools/convert4qmc.cpp:
```diff
@@ -55,10 +55,23 @@ int main(int argc, char **argv)
+  token = prefix.substr(0, pos);
+  prefix.erase(0, pos + delimiter.length());
+  prefix=token;
+  std::cout << "PREFIX="<<prefix << std::endl;
```
Friendlier formatting "Using " << prefix << " to name output files"

anbenali replied:
Done

prckent commented on src/QMCTools/QMCGaussianParserBase.cpp:
```diff
@@ -1566,6 +1575,7 @@ void QMCGaussianParserBase::createGridNode(int argc, char** argv)
             if(a == "-nojastrow")
```
Remove the nojastrow option and *always* create both files .wfj and .wfnoj

I don't think space can be a concern due to the hdf5 options.

anbenali replied:
Any Update on this? @ye-luo @prckent ? 
Since you are the only two who cared, what do you prefer? On file? or two files. If two files, which one goes in the Input.

ye-luo replied:
I prefer it simple and better if backward compatiable. Just one file. noj can be created manually by users in 2 secs.

ye-luo commented:
Can we have just one prefix.wfs.xml instead of two files with different names? I just feel it unnecessary and harder for scripting or making workflow. If the xml file is large, we make things even worse now.
For the particle set file, it is describing the partcle sets (elecs, ions, cell) not just the structure. I would prefer not changing it.

prckent commented:
@ye-luo  

Wavefunctions: I have decided that we are going to strongly encourage people to run checks with no jastrow. As the inputs currently work I think this requires two files. We will also eventually be producing QMCPACK inputs to handle these files so that we always produce 

Structure: The important contents of this file is the atom positions or structure. Other ideas are welcome, but the cryptic, non-user friendly, and code specific "particle set" terminology is not going to be used.

prckent commented:
General comment: This converter is the first thing that new users of QMCPACK are likely to encounter. e.g. "Can I run a water molecule?". Suggestions for updates should be viewed through this lens.

prckent commented:
@ye-luo Misunderstood your comment. Good idea. @anbenali Do you think these will be too big in the large molecule case or can the HDF5 input also get the structure? If the size/complexity is no unreasonable, why not a single .wfj.xml for the wavefunction with jastrow and a .wfnoj.xml for the determinant(s) only file?

ye-luo commented:
The only difference between .wfj.xml and .wfnoj.xml is only a few Jastorw blocks. The major contents (determinants dominant) are identical. I have seen files as large as 60MB for large molecules.
I feel it unnecessary to generate both files and users can very easily copy the file and delete the jastrow blocks if they need a noJ wavefunction for testing. It is a no pain step for user and they should verify the file contents rather than blindly trust the file name.

markdewing commented on tests/converter/Testing/Temporary/CTestCostData.txt:
```diff
@@ -0,0 +1 @@
+---
```
This file should not be checked in.

anbenali replied:
Done

markdewing commented on tests/converter/Testing/Temporary/LastTest.log:
```diff
@@ -0,0 +1,3 @@
+Start testing: Dec 04 22:13 CST
```
This file should not be checked in.

anbenali replied:
Done

ye-luo commented on src/QMCTools/convert4qmc.cpp:
```diff
@@ -55,10 +55,23 @@ int main(int argc, char **argv)
   std::string psi_tag("psi0");
   std::string ion_tag("ion0");
-  std::string prefix("sample");
+  std::string prefix(argv[2]);
```
This is not safe, based on the help info. The first argument can be omitted and argv[2] is null.

anbenali replied:
The first argument omitted never works. 
Maybe it is time to remove that option and make a clean fail if you don't provide the first argument.

ye-luo replied:
I tried it. It works on our test cases.

anbenali replied:
In all cases, I "secured" it for now.

anbenali commented:
Remaining issue to discuss:
- wfs VS wfj +wfnoj

I personally prefer only one file but I do not care that much about it to spend time on the issue. I will go with majority.

prckent commented:
Please don't print the basis functions on stdout.

anbenali commented:
From @prckent : I addressed all your requests:

>>    Support -gamess , deprecate -gamessAscii.
  Done
>>    Don't print the basis functions on stdout.
  Since this is useful when introducing a new converter, I added a tag -debug that will print it. By default it is not printed anymore.

>>    Don't write the .combo files
only printed with the -debug
 >>  Presumably these can become huge? Does anyone use them?
No idea who uses them but they can be useful so accessible through -debug
   
>> Write wfj and wjfnoj
  all set of combinations are written by default.

>>  We can add an option to only write one later
I do need this one as I really do not want to have multiple files laying around. by using the tag "-writewfjonly" only the requested files will be created.
    
>> Remove all the gpu="no" tags. This clutters the file
        The code will stop if anything non-supported is reached. Sadly, the code does not support cpu fallbacks.

Commented for now.

prckent commented:
@anbenali Thanks. I am going to merge this so that the nightly tests can start seeing these updates.  Revisions to the example inputs can be addressed in a later issue/PR.
